### PR TITLE
EWPP-4784: Do not add LACO widget if all coverage options are disabled.

### PR DESCRIPTION
**Pull request by intelektron**
### Description

The LACO widget will not be added to the page, if all coverage options are set to "false".

### CODE REVIEW

File: modules/oe_webtools_laco_widget/oe_webtools_laco_widget.module [hunk 1]
```diff
@@ -16,11 +16,21 @@ function oe_webtools_laco_widget_page_bottom(array &$page_bottom) {
   $logger = \Drupal::logger('oe_webtools_laco_widget');
   $config = \Drupal::config('oe_webtools_laco_widget.settings');
 
+  $cache = CacheableMetadata::createFromRenderArray($page_bottom);
```
Comment (intelektron) [inline modules/oe_webtools_laco_widget/oe_webtools_laco_widget.module:19]: I moved this code up, so it also works for the "guard" conditions.
Without it, enabling LACO back requires a cache clear.

File: modules/oe_webtools_analytics/modules/oe_webtools_analytics_rules/src/RuleMatcher.php [hunk 2]
```diff
@@ -91,14 +91,14 @@ public function __construct(AliasManagerInterface $aliasManager, CacheBackendInt
   /**
    * {@inheritdoc}
    */
-  public function getMatchingSection(string $path = NULL): ?string {
```
Comment (intelektron) [inline modules/oe_webtools_analytics/modules/oe_webtools_analytics_rules/src/RuleMatcher.php:94]: I am aware that these changes were already introduced by @22Alexandra in EWPP-4782, but I put it in the separate commit, so it is easier to rebase.

File: modules/oe_webtools_laco_widget/oe_webtools_laco_widget.module [hunk 3]
```diff
@@ -16,11 +16,21 @@ function oe_webtools_laco_widget_page_bottom(array &$page_bottom) {
...
+  $cache->applyTo($page_bottom);
+
   // Do not add LACO widget related JSON if it is disabled in the config.
   if (!$config->get('enabled')) {
     return;
   }
 
+  // Do not add LACO widget related JSON if both coverage options are disabled.
+  $coverage = $config->get('coverage');
+  if (($coverage['document'] === 'false') && ($coverage['page'] === 'false')) {
```
Comment (22Alexandra) [inline modules/oe_webtools_laco_widget/oe_webtools_laco_widget.module:30]: Are the extra parenthesis needed?
```suggestion
  if ($coverage['document'] === 'false' && $coverage['page'] === 'false') {
```

**Review by sergepavle — APPROVED** [review]

**Review by 22Alexandra — APPROVED** [review]

**Review by sergepavle — APPROVED** [review]
